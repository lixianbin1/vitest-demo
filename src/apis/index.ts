import require from './require'

const auto = import.meta.env.VITE_APP_Auto
const network = import.meta.env.VITE_APP_NetWork

interface HomeResponse {
  name: string;
}
export const HomeJson = function (params?:object):Promise<HomeResponse> {
  return require({
    url: `${network}/${auto}/text.json`,
    method: 'get',
    params,
  })
}

interface LoginResponse {
  userToken: string;
}
export const Userlogin = function (params?:object):Promise<LoginResponse> {
  return require({
    url: `${network}/${auto}/login.json`,
    method: 'post',
    params,
  })
}