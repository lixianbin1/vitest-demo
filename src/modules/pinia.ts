import { App } from 'vue'
import { createPinia } from 'pinia'

export const PiniaStore = createPinia()
export const install = (app:App) => {
  app.use(PiniaStore)
}
export default install