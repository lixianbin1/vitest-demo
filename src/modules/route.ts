import { App } from 'vue'
import { createRouter,createWebHashHistory } from 'vue-router'
import { setupLayouts } from 'virtual:generated-layouts'
import generatedRoutes from '~pages'

export const routes = setupLayouts(generatedRoutes)
export const router = createRouter({
  history: createWebHashHistory(),
  routes,
})
export const install = (app:App) => {
  app.use(router)
}
export default install