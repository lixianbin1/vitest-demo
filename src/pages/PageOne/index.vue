<template>
  <div class="head">
    <h1>Page One</h1>
    <h2>{{data}}</h2>
  </div>
</template>

<script lang="ts" setup>
import { HomeJson } from '@/apis';

const data = ref({})
const goHome = () => {
  console.log(HomeJson)
  HomeJson().then(json=>{
    data.value = json
    console.log(data)
  }).catch(err=>{
    console.log(err)
  })
}

goHome()
</script>

<style scoped>

</style>