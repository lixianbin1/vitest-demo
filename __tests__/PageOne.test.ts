import {describe,expect,test,vi} from 'vitest'
import {render,screen} from '@testing-library/vue'
import { mount } from '@vue/test-utils';
import PageOne from '../src/pages/PageOne/index.vue'
import {i18n} from '../src/modules/i18n'
import {router} from '../src/modules/route'
import { HomeJson } from '@/apis'
  
// 模拟HomeJson函数  
vi.mock('@/apis', () => ({  
  HomeJson: vi.fn().mockResolvedValue('mocked data'),  
}));  

describe('PageOne',()=>{
  test('PageOne快照',async()=>{
    const { container } = render(PageOne,{
      global: {
        plugins:[i18n,router]
      }
    })
    await new Promise(resolve => setImmediate(resolve));
    expect(container).toMatchSnapshot()
  })
  
  test('渲染标题和数据', async () => {
    // render(PageOne);  
    await new Promise(resolve => setImmediate(resolve)); // 等待异步操作完成  
  
    // 检查标题 "Page One" 是否存在  
    const headingOne = screen.getByText('Page One');  
    expect(headingOne).toBeDefined(); // 确保元素被定义  
    expect(headingOne.tagName).toBe('H1'); // 检查标签名  
  
    // 检查数据 "mocked data" 是否渲染  
    const headingTwoData = screen.getByText('mocked data');  
    expect(headingTwoData).toBeDefined();  
    expect(headingTwoData.tagName).toBe('H2'); 
  
    // 检查HomeJson是否被调用  
    expect(HomeJson).toHaveBeenCalled();  
  });  

})
