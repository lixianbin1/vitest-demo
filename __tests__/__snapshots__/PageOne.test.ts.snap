// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageOne > PageOne快照 1`] = `
<div>
  <div
    class="head"
  >
    <h1>
      Page One
    </h1>
    <h2>
      mocked data
    </h2>
  </div>
</div>
`;
